gateway/reverse_proxy.go

```diff
@@ -91,7 +89,12 @@ func urlFromService(spec *APISpec) (*apidef.HostList, error) {
 			return spec.LastGoodHostList, nil
 		}
 
-		ServiceCache.Set(spec.APIID, data, cache.DefaultExpiration)
+		ttl, cacheEnabled := spec.Proxy.ServiceDiscovery.CacheOptions()
+		if !cacheEnabled {
+			ttl = 0 // use gateway default cache time to live.
+		}
+		gw.ServiceCache.Set(spec.APIID, data, ttl)
```

Comment:
I find it confusing why are we setting the cache here if `cacheEnabled = false`

Reply:
@tbuchaillot does the comment two lines above not give enough of a hint? cacheEnabled comes from api spec, but we also have gateway (service_discovery.default_cache_timeout). The consistent way is to have cache:{enabled, timeout}, which is what we're doing here, but we still need to respect GW settings.